src/api/utils/resourceHookFactories.ts

```diff
@@ -62,22 +62,40 @@ export const createUseMutation = <Input, Result>(
   };
 };
 
-export const createUseMutationDelete = (
-  key: string[],
-  url: string,
-  fetchOptions?: RequestInit,
+interface MutationDeleteOrPutProps {
+  key: string[];
+  url: string;
+  fetchOptions?: RequestInit;
   mutationOptions?: Omit<
     UseMutationOptions<null, unknown, number, unknown>,
     'mutationFn'
-  >
+  >;
+}
+
+export const createUseMutationDelete = (
+  props: MutationDeleteOrPutProps
```

Comment:
Curious to hear @djbusstop feedback on this refactor.

Reply:
I am fine with it!